types/rule_match.go

```diff
@@ -47,4 +47,6 @@ type MatchedRule interface {
 	AuditLog() string
 
 	ErrorLog() string
+
+	// TODO(4.x): Add Log()
```

Comment:
Do you mind elaborating why?

Reply:
I remember that with https://github.com/corazawaf/coraza/pull/848 we agreed that `Log()` was indeed helpful to be exported, but then to avoid breaking the Coraza v3.* API adding a Log() method to the MatchedRule interface we provided the workaround via the type assertion. So I guess that the proper outcome is to have it properly exported.
See also https://github.com/corazawaf/coraza/blob/e29a849961c1688a7c0700b139262445a098aba3/internal/corazawaf/transaction.go#L1499-L1503